pkg/reconciler/cache/replication/replication_controller.go

```diff
@@ -62,20 +67,57 @@ func NewController(
 ) (*controller, error) {
 	queue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), ControllerName)
 	c := &controller{
-		shardName:              shardName,
-		queue:                  queue,
-		dynamicCacheClient:     dynamicCacheClient,
-		dynamicLocalClient:     dynamicLocalClient,
-		localApiExportLister:   localApiExportInformer.Lister(),
-		cacheApiExportsIndexer: cacheApiExportInformer.Informer().GetIndexer(),
+		shardName: shardName,
+		queue:     queue,
+
+		getCachedAPIExport: func(shardName string, cluster logicalcluster.Name, name string) (*apisv1alpha1.APIExport, error) {
```

Comment:
What was wrong with the previous approach? Is is about testability? 
At first sight, the proposed solution requires more code.

Reply:
clarified with @stevekuznetsov offline. I'm going to take over this PR and adapt it to the recent changes I've made to the replication controller.